tests/Insightly/Listeners/UnblockOpportunityTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Insightly\Listeners;
+
+use App\Domain\Integrations\Events\IntegrationUnblocked;
+use App\Domain\Integrations\Integration;
+use App\Domain\Integrations\IntegrationPartnerStatus;
+use App\Domain\Integrations\IntegrationStatus;
+use App\Domain\Integrations\IntegrationType;
+use App\Domain\Integrations\Repositories\IntegrationRepository;
+use App\Insightly\InsightlyMapping;
+use App\Insightly\Listeners\UnblockOpportunity;
+use App\Insightly\Objects\OpportunityState;
+use App\Insightly\Repositories\InsightlyMappingRepository;
+use App\Insightly\Resources\ResourceType;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+use Tests\MockInsightlyClient;
+
+final class UnblockOpportunityTest extends TestCase
+{
+    use MockInsightlyClient;
+
+    private InsightlyMappingRepository&MockObject $insightlyMappingRepository;
+
+    private IntegrationRepository&MockObject $integrationRepository;
+
+    private UnblockOpportunity $unblockOpportunity;
+
+    protected function setUp(): void
+    {
+        $this->insightlyMappingRepository = $this->createMock(InsightlyMappingRepository::class);
+
+        $this->integrationRepository = $this->createMock(IntegrationRepository::class);
+
+        $this->mockCrmClient();
+
+        $this->unblockOpportunity = new UnblockOpportunity(
+            $this->insightlyClient,
```

Comment:
I seem to miss the asserts/expects for the mocked Insightly client?

Reply:
Isn't that handled by `MockInsightlyClient` in https://github.com/cultuurnet/publiq-platform/blob/1541440b38978063079684b9761d3eca202d464f/tests/MockInsightlyClient.php#L38?

 There can only be an `opportunityResource`, which has an `expects()` if the `InsightlyClient` was called

Reply:
There is no `expects` method added, only a `method` and `willReturn`.
See `ActivateProjectTest` for an example how the mocked resources are then used with an `expects`

Reply:
Please discard this remark, the `expects` on the `opportunityResource` itself is enough. There is indeed no added value in calling the `expects` on `insightlyClient`.